src/features/device-heartbeat/index.ts

```diff
@@ -436,7 +442,10 @@ export class DeviceOnlineStateManager extends EventEmitter<{
 		// If redis still has a valid message about the device being online we can avoid reaching to the DB...
 		if (
 			previousDeviceOnlineState == null ||
-			previousDeviceOnlineState.currentState !== DeviceOnlineStates.Online
+			previousDeviceOnlineState.currentState !== DeviceOnlineStates.Online ||
```

Comment:
Shouldn't we make it a continuous  cache busting mechanism? I see that the redis cache is updated with new expiration, should we maybe rethink this continuous update, or should we allow it to get stale after the expiration has run out, access the DB and write the redis with DB synced data?

Reply:
Can you expand on how a continuous cache busting mechanism would work?
The primary point of this "cache" is to avoid reaching continuously to the DB

Theoretically with #1336 we should no longer get into this inconsistent state, since we will only be updating Redis only after we are sure that the DB write was persisted.
The fact that we only got such reports on support after the PG upgrade sounds like going out of sync is already not that common.
In case though that we are missing more code paths that can lead to an inconsistent state, we can still as a hotfix change the cache busting key to fix it for customers while trying to find the code path that allowed going out of sync at the first place.
An other approach I was thinking was for example to have a counter and persist the write to the DB on every Nth heartbeat or something. That would conceal though possible paths that can lead to inconsistent states, so I preferred to leave that as a last resort.

Reply:
@fisehara here is an alternative w/ a TTL approach #1339